### PR TITLE
add calicoctl patch docs

### PR DESCRIPTION
**Pull request by syscll**
## Description

Add docs for new `calicoctl patch` command https://github.com/projectcalico/calicoctl/pull/2054

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit 7494d101b6f1a553f6b844a5f8d8d5bf554fae45

https://deploy-preview-2856--calico.netlify.com

File: master/reference/calicoctl/patch.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  
+  When patching resources by type, only a single type may be specified at a
+  time.  The name is required along with any and other identifiers required to
+  uniquely identify a resource of the specified type.
+```
+{: .no-select-button}
+
+### Examples
+
+1. Patch and ipPool to enable outgoing NAT:
```
Comment (caseydavenport) [inline master/reference/calicoctl/patch.md]: typo: "Patch and"

File: master/reference/calicoctl/patch.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+   ```bash
+   calicoctl patch ippool ippool1 -p '{"spec":{"natOutgoing": true}}'
+   ```
+
+   Results indicate that 8 resources were successfully applied
```
Comment (caseydavenport) [inline master/reference/calicoctl/patch.md]: typo: 8 resources applied

File: master/reference/calicoctl/patch.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                          [default: /etc/calico/calicoctl.cfg]
+```
+{: .no-select-button}
+
+## See also
+
+-  [Installing calicoctl]({{site.baseurl}}/{{page.version}}/getting-started/calicoctl/install)
+-  [Resources]({{site.baseurl}}/{{page.version}}/reference/resources/) for details on all valid resources, including file format
+   and schema
+-  [NetworkPolicy]({{site.baseurl}}/{{page.version}}/reference/resources/networkpolicy) for details on the {{site.prodname}} selector-based policy model
```
Comment (caseydavenport) [inline master/reference/calicoctl/patch.md]: Is network policy relevant as a see also?

**Comment by syscll** [comment]
Have followed the instructione [here](https://github.com/projectcalico/calico/blob/master/CONTRIBUTING_DOCS.md#creating-new-pages) for creating a new page, but it doesn't seem to pass CI.

File: master/reference/calicoctl/patch.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
+---
+title: calicoctl patch
+canonical_url: 'https://docs.projectcalico.org/master/reference/calicoctl/commands/patch'
```
Comment (caseydavenport) [inline master/reference/calicoctl/patch.md]: I think this is what is causing your CI to fail - it should pass once merged (this URL doesn't exist yet).

File: master/reference/calicoctl/patch.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+---
+
+This sections describes the `calicoctl patch` command.
+
+Read the [calicoctl command line interface user reference]({{site.baseurl}}/{{page.version}}/reference/calicoctl/)
+for a full list of calicoctl commands.
+
+> **Note**: The available actions for a specific resource type may be
```
Comment (caseydavenport) [inline master/reference/calicoctl/patch.md]: I think we can remove this note - patch, at least, applies to both.

File: master/reference/calicoctl/patch.md (hunk 4 above)
Comment (syscll) [reply master/reference/calicoctl/patch.md]: That's what I thought, but just wanted to check!

File: master/reference/calicoctl/patch.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  
+  When patching resources by type, only a single type may be specified at a
+  time.  The name is required along with any and other identifiers required to
+  uniquely identify a resource of the specified type.
+```
+{: .no-select-button}
+
+### Examples
+
+1. Patch an ipPool to enable outgoing NAT:
```
Comment (caseydavenport) [inline master/reference/calicoctl/patch.md]: `ipPool` -> `IP pool` (we should use sentence case)

**Pull request merged**